### PR TITLE
Pin for docker-ce

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one- Not for changelog (changelog entry is not required)
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
See https://github.com/moby/moby/issues/45770#issuecomment-1618255130

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
This is an automated comment for commit cb26b4bf8517f14933666edac7f3ea167abd304c with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51743/cb26b4bf8517f14933666edac7f3ea167abd304c/ci_running.html)
The overall status of the commit is 🔴 failure

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🔴 failure</td></tr>
</table>

File: docker/test/integration/runner/Dockerfile [hunk 1]
```diff
@@ -47,6 +47,8 @@ ENV TZ=Etc/UTC
 RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
 
 ENV DOCKER_CHANNEL stable
+# Unpin the docker version after the release 24.0.3 is released
```
Comment (alexey-milovidov) [inline docker/test/integration/runner/Dockerfile:50]: I see only a comment, but where is the pin?

**Comment by alexey-milovidov** [comment]
Ok, but we should always pin everything - every package, every docker image, etc.
So, it will not be updated under our feet.

File: docker/test/integration/runner/Dockerfile (hunk 1 above)
Comment (Felixoid) [reply docker/test/integration/runner/Dockerfile:50]: oh my, you are right

**Comment by Felixoid** [comment]
https://github.com/ClickHouse/ClickHouse/issues/51682#issuecomment-1619121885

I prefer updating in a rolling manner VS completely rotten infrastructure.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by Felixoid** [comment]
Should we merge it, or spend another round on running it on top of master?

**Comment by Felixoid** [comment]
Let's backport it to the affected release as well

**Pull request merged**